cranelift-filetests/src/function_runner.rs

```diff
@@ -76,13 +179,175 @@ impl FunctionRunner {
+                ConstantData::default().append(1),
+                true,
+            ),
+            FunctionRunnerAction::TestWithResult { operator, expected } => {
+                let expected = if expected.len() < (ret_value_type.bytes() as usize) {
+                    expected.clone().expand_to(ret_value_type.bytes() as usize)
+                } else {
+                    expected.clone()
+                };
+
+                Self::check(code_page, ret_value_type, *operator, expected, false)
+            }
+        }
+    }
+}
+
+fn return_type(func: &Function) -> Result<types::Type, String> {
+    let normal_rets: Vec<&AbiParam> = func
+        .signature
+        .returns
+        .iter()
+        .filter(|r| r.purpose == ArgumentPurpose::Normal)
+        .collect();
+    if normal_rets.len() > 2 || normal_rets.len() == 0 {
+        return Err(String::from("Function return type is not supported"));
+    }
+    if normal_rets.len() == 2 {
+        // Only two i64 return arguments supported and they are evaluated as a i128.
+        let both_i64 = normal_rets
+            .into_iter()
+            .map(|p| p.value_type)
+            .all(|ty| ty == types::I64);
+
+        if !both_i64 {
+            return Err(String::from(
+                "Function return type with two parameters can only return i64, i64.",
+            ));
+        }
+
+        return Ok(types::I128);
+    }
+    return Ok(normal_rets[0].value_type);
+}
+
+impl FunctionRunnerBuilder {
+    /// Create a new function runner builder with the specified function.
+    pub fn new(function: Function) -> Self {
+        Self {
+            action: FunctionRunnerAction::Test,
+            function: function,
+            isa: None,
+        }
+    }
+
+    /// Use the specified action.
+    pub fn with_action(mut self, action: FunctionRunnerAction) -> Self {
+        self.action = action;
+        self
+    }
+
+    /// Parse and use the specified action.
+    pub fn with_action_text(mut self, action: &str) -> Self {
+        self.action = Self::parse_action(action, return_type(&self.function).unwrap());
+        self
+    }
+
+    /// Use the specified ISA (must be the host machine's ISA).
+    pub fn with_isa(mut self, isa: Box<dyn TargetIsa>) -> Self {
+        self.isa = Some(isa);
+        self
+    }
+
+    /// Use the host ISA with the specified flags.
+    pub fn with_host_isa(self, flags: settings::Flags) -> Self {
+        let builder = host_isa_builder().expect("Unable to build a TargetIsa for the current host");
+        self.with_isa(builder.finish(flags))
+    }
+
+    /// Use the default host ISA.
+    pub fn with_default_host_isa(self) -> Self {
+        let flags = settings::Flags::new(settings::builder());
+        self.with_host_isa(flags)
+    }
+
+    /// Build a function runner from an action, a function and the ISA to run on.
+    pub fn finish(self) -> FunctionRunner {
+        FunctionRunner::new(self.function, self.isa.unwrap(), self.action)
+    }
```

Comment:
Nit: elsewhere in the project we are calling this `build`.

Reply:
`cranelift-codegen/src/isa/mod.rs` calls it `finish`.

https://github.com/bytecodealliance/cranelift/blob/da996fe5ee4ea32a57156ba93f6813b052964183/cranelift-codegen/src/isa/mod.rs#L151

Reply:
Yeah, in this particular case it's this file which is not consistent with the Rust style, so we should fix it (later).